datalad/distribution/add.py

```diff
@@ -0,0 +1,287 @@
+    data from. If no `path` argument is provided in this case, the content will
+    be obtained from the source location and a default local name, derived from
+    the source location will be generated. Alternatively, an explicit `path`
+    can be given to override the default.
+
+    If more than one `path` argument and a source location are provided, the
+    `path` arguments will be sequentially used to complete the source URL/path
+    (be means of concatenation), and an attempt is made to obtain data from
+    those locations.
+
+
+    || REFLOW >>
+    By default all files are added to the dataset's :term:`annex`, i.e. only
+    their content identity and availability information is tracked with Git.
+    This results in lightweight datasets. If desired, the [PY: `to_git`
+    PY][CMD: --to-git CMD] flag can be used to tell datalad to inject files
+    directly into Git. While this is not recommended for binary data or large
+    files, it can be used for source code and meta-data to be able to benefit
+    from Git's track and merge capabilities. Files checked directly into Git
+    are always and unconditionally available immediately after installation of
+    a dataset.
+    << REFLOW ||
+
+    .. note::
+      Power-user info: This command uses :command:`git annex add`, :command:`git annex addurl`, or
+      :command:`git add` to incorporate new dataset content.
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            metavar='PATH',
+            doc="""specify the dataset to perform the add operation on.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar='PATH',
+            doc="""path/name of the component to be added. The component
+            must either exist on the filesystem already, or a `source`
+            has to be provided.""",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        source=Parameter(
+            args=("-s", "--source",),
+            metavar='URL/PATH',
+            doc="url or local path of the to be added component's source",
+            action="append",
+            constraints=EnsureStr() | EnsureNone()),
+        to_git=Parameter(
+            args=("--to-git",),
+            action='store_true',
+            doc="""flag whether to add data directly to Git, instead of
+            tracking data identity only.  Usually this is not desired,
+            as it inflates dataset sizes and impacts flexibility of data
+            transport"""),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_add_opts=annex_add_opts)
+
```

Comment:
@bpoldrack @yarikoptic I guess we should decide how/if we want to support saving changing in each interface. Aggregate does it, this one doesn't. I see boths pros and cons for both. Adding this option makes things more convoluted, but when ON by default we could tailor tthe resulting commits to be more meaningful -- compared to a `save -a -r` after some time. Sure thing there should always be a way to _not_ auto-save things.